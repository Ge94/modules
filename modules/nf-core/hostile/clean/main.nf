process HOSTILE_CLEAN {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/hostile:1.1.0--pyhdfd78af_0':
        'biocontainers/hostile:1.1.0--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(reads)
    path(reference)

    output:
    tuple val(meta), path("cleaned_reads/"), emit: cleaned_reads
    path "versions.yml"                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def reads_cmd = meta.single_end ? "--fastq1 ${reads.sort()[0]}" : "--fastq1 ${reads.sort()[0]} --fastq2 ${reads.sort()[1]}"
    """
    export HOSTILE_CACHE_DIR=${reference}
    mkdir cleaned_reads/

    hostile \\
        clean \\
        $args \\
        --threads $task.cpus \\
        --out-dir cleaned_reads/ \\
        --reorder \\ # For reproducability
        --offline \\ # This module requires input reference files so never download in this process
        $reads_cmd

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hostile: \$(hostile --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir cleaned_reads/
    touch ${prefix}_r1.fastq.gz
    touch ${prefix}_r2.fastq.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hostile: \$(hostile --version)
    END_VERSIONS
    """
}
