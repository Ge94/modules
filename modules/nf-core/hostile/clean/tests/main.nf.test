nextflow_process {

    name "Test Process HOSTILE_CLEAN"
    script "../main.nf"
    process "HOSTILE_CLEAN"
    config './nextflow.config'

    tag "modules"
    tag "modules_nfcore"
    tag "hostile"
    tag "hostile/clean"

    setup {
        run("BOWTIE2_BUILD") {
            script "../../../bowtie2/build/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("sarscov2 - fastq - single-end") {
        when {
            params {
                module_args = "--aligner bowtie2"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:true ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                    ],
                ]
                input[1] = BOWTIE2_BUILD.out.index.map{it[1]}
                input[2] = 'genome'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                        process.out.cleaned_reads,
                        process.out.versions,
                        process.out.log[0][1].readLines().contains('"reads_removed": 200')
                    ).match() }
            )
        }

    }


    test("sarscov2 - fastq - paired-end") {
        when {
            params {
                module_args = ""
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true),
                    ],
                ]
                input[1] = BOWTIE2_BUILD.out.index.map{it[1]}
                input[2] = 'genome'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                        process.out.cleaned_reads,
                        process.out.versions,
                        process.out.log[0][1].readLines().contains('"reads_removed": 200')
                    ).match() }
            )
        }

    }

    test("sarscov2 - fastq - single-end - stub") {

        options "-stub"

        when {
            params {
                module_args = ""
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:true ], // meta map
                    [file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)],
                ]
                input[1] = BOWTIE2_BUILD.out.index.map{it[1]}
                input[2] = 'genome'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
