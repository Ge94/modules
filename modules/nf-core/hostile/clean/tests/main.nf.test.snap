{
    "sarscov2 - fastq - single-end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.clean_1.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test.clean_2.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,2a9f7b2ced8f470e6cd026b68aad9608"
                ],
                "cleaned_reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.clean_1.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test.clean_2.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2a9f7b2ced8f470e6cd026b68aad9608"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T13:58:30.020635321"
    },
    "sarscov2 - fastq - paired-end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.clean_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_2.clean_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.log:md5,aed1f2fd60cae7c14b60632cb55c6657"
                    ]
                ],
                "2": [
                    "versions.yml:md5,2a9f7b2ced8f470e6cd026b68aad9608"
                ],
                "cleaned_reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.clean_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_2.clean_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.log:md5,aed1f2fd60cae7c14b60632cb55c6657"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2a9f7b2ced8f470e6cd026b68aad9608"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T13:58:10.034488637"
    },
    "sarscov2 - fastq - single-end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_1.clean.fastq.gz:md5,9f846701606f9aaf548bbd0d0fda8ec1"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.log:md5,7c05d265747f480b2728e01830d601cd"
                    ]
                ],
                "2": [
                    "versions.yml:md5,2a9f7b2ced8f470e6cd026b68aad9608"
                ],
                "cleaned_reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_1.clean.fastq.gz:md5,9f846701606f9aaf548bbd0d0fda8ec1"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.log:md5,7c05d265747f480b2728e01830d601cd"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2a9f7b2ced8f470e6cd026b68aad9608"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-13T13:57:45.849860634"
    }
}